enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+
+## Summary
+
+The purpose of this enhancement is to decouple the cnf-tests from the test suites of the external CNF projects. Additionally, as this change affects the way tests are run, we propose rewriting the cnf-tests container image, which is a supported product for running the latency tests.
+## Motivation
+
+In the cnf-tests, all three test suites (configsuite, e2esuite, valdiationsuite) are importing
+external test suites (outside of the cnf-feauters-deploy repository) with a [blank identifier](https://golangbyexample.com/blank-identifier-import-golang/). This is done to compile their tests into one binary. For example, in the e2esuite, we import the following:
+
+```go
+import (
+    _ "github.com/k8snetworkplumbingwg/sriov-network-operator/test/conformance/tests"
+    _ "github.com/metallb/metallb-operator/test/e2e/functional/tests"
+    _ "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance" 
+    _ "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency" 
+    _ "github.com/openshift/ptp-operator/test/conformance/serial"
+)
+```
+
+To prevent conflicts due to the numerous dependencies in the test suites, we want to decouple the external test suites by introducing a script that invokes the required test suites as git-submodules under the cnf-tests/submodules path.
+### Note
+1. In this proposal, we refer to each external test suite as a CNF component.
+2. Some dependencies will still be required in the components as there are tests that combine multiple features.
+
+
+This change will also allow us to pin the external repositories to the latest branch, ensuring that we always test the latest tests without the need for manual commit hash updates. We will include an option to set an environment variable that specifies the commit ref for each repo.
+
+For detailed information about the changes in the cnf-tests container, please refer to the section titled [Rewrite the cnf-tests container](#rewrite-the-cnf-tests-container).
+
+### Goals
+
+1. Stop vendoring the external test suites and run the tests directly from the projects.
+2. Maintain the same API for running the tests.
+3. Preserve the same testing report output.
+4. Refactor the cnf-tests container to contain only the latency-tests.
+
+### Non-Goals
+
+1. Rename existing tests to a certain standard.
+2. Change the scope of the tests being run.
+
+## Proposal
+### User Stories
+#### Story 1
+
+As a developer, I want to run the cnf-tests suites so that I can validate my CNF feature works properly.
+
+#### Story 2
+As an Openshift administrator, I want to run the latency tests so that
+I can benchmark different CNF features.
+
+### Workflow Description
+
+This change will not affect the current API, including parameters/environment variables used to invoke the cnf-tests: `TEST_SUITES`, `FEATURES`, `FOCUS_TESTS`, `SKIP_TESTS`, `GINKGO_PARAMS`, `FAIL_FAST`, and `TESTS_REPORTS_PATH`. 
```

Comment:
how about discovery mode?

Reply:
And how about the deployment mechanism that currently leverages some of the same env variables?